### DISCUSSION TITLE
Upload GGUF not working

### DISCUSSION THREAD

**Discussion by Kaszebe** [open-webui/open-webui#3908] [Q&A] [answered]
I have VLLM and Open WebUI installed on my PopOS server. 

I go to Models > Upload a GGUF Model and then select the model. I then click the upload button and an Upload Progress  bar appears. However, nothing gets uploaded.

This may be a noob question but do I need to somehow "connect" VLLM to Open WebUI? I'm fairly certain I do but have no idea how to do it. ChatGPT was not very helpful.

thank you

**Comment by justinh-rahb** [comment, marked as answer]
GGUF upload is for Ollama backends, won't work for vLLM.

**Reply by Kaszebe** [reply]
I recall reading somewhere that the latest version of vLLM supports GGUF now.

**Reply by justinh-rahb** [reply]
Be that as it may, the GGUF uploader in WebUI only connects with Ollama servers.

**Reply by Kaszebe** [reply]
Ah. Is there any way to import a GGUF that I downloaded off HuggingFace?

**Reply by Dihelson** [reply]
Yes, I have many GGUF models I downloaded for using on LM Studio. I discovered that I can use them on Open WebUI. You only need to access: Settings, Admin settings, Models, Experimental. Click on "show". On upload a GGUF model, on file mode, it's displayed a new tab called: Click here to select". Click on it. A window will open, and you only need to point to your GGUF file. After that click on upload file. Open WebUI will display a circle signaling that it's converting the file to the accepted format, and placing on the correct folder. After that, which can take some minutes, Open WebUI will return to normal state. After that, you can already access your new model, available. I have converted many of what I used on Hugging Face with complete success. Good luck 🤞.

**Reply by Kaszebe** [reply]
Hi, I did all of that and the progress bar does not move forward. Does this mean that VLLM is not properly connected to Open WebUI? 

I had to resort to ChatGPT to help me—and it's suggesting that because I'm running VLLM in a container and Open WebUI in a container, they're not able to communicate with each other. I'm a bit of a n00b when it comes to Linux and I don't know enough to tell ChatGPT it's full of shit—or to listen to its advice (which I'm not comfortable doing at this point in time until I get a human being to let me know if I'm on the right track).

**Reply by justinh-rahb** [reply]
It's not full of it. We already stated what you are asking is not possible.

**Reply by Kaszebe** [reply]
Dihelson just said it is possible.

**Reply by justinh-rahb** [reply]
One or both of you is confused then. What Dihelson did was upload a GGUF to Ollama. This is the only backend we use that our GGUF uploader can interact with.

**Reply by Kaszebe** [reply]
Yeah I'm beyond confused. I spoke with someone who said Open WebUI is his go-to front end for VLLM. I installed Open WebUI and I really like the features. I was hoping to find a way to connect VLLM to Open WebUI and use my GGUFS that I downloaded from HuggingFace (that currently reside in my /models folder in OobaBooga).

Technically speaking, I have 88GB of VRAM to play with and that will be increasing in a week or two by another 48GB. With that being said, I can also run EXL2 files as well.

**Reply by justinh-rahb** [reply]
vLLM doesn't even support GGUF model formats, nor EXL2. You can use whatever model quantization formats that it does support including **GPTQ** and **AWQ**, you just need to manage downloading and loading them yourself. WebUI connects to vLLM endpoints for inference, but it does not **manage** them, this seems to be where the confusion lies.

**Reply by Kaszebe** [reply]
Ah, so I load the files into VLLM and use WebUI as the front end to perform inference?

Can you recommend a good back end for someone with 4 GPUs (and plans to ultimately have ten 3090s)? I looked into Aphrodite...but there was some reason that made me question if it was right for me. I chose VLLM because I heard it does a great job of managing multiple GPUs. 

Oobabooga is not doing a great job with managing my GPUs. No matter how I try to modify the tensor_split (or row_split, which does it automatically), there are several ~65GB models (GGUF, EXL2) that will not load into my 88GB of VRAM.

This is for work purposes (trying to learn how to fine-tune because ChatGPT/etc are woefully inadequate for my purposes) as well as educational purposes. And I'm finding that AI is becoming less of a hobby and more of an obsession to learn as much as I can.
